camayoc/tests/qcs/test_profiles.py

```diff
@@ -113,31 +168,71 @@ def test_negative_create_missing_data(cleanup):
             b) hosts
             c) credential id's
     :expectedresults: Error is thrown and no new profile is created.
-    :caseautomation: notautomated
+    :caseautomation: automated
     """
-    pass
+    cred = HostCredential(client=shared_client, password=uuid4())
+    cred.create()
+    cleanup.append(cred)
+    profile = NetworkProfile(
+        client=api.Client(response_handler=api.echo_handler),
+        hosts=['localhost'],
+        credential_ids=[cred._id],
+    )
 
+    # remove hosts from fields
+    delattr(profile, data)
+    create_response = profile.create()
+    assert create_response.status_code in range(400, 500)
+    read_response = profile.create()
+    assert read_response.status_code in range(400, 500)
 
-@pytest.mark.skip
-def test_negative_create_invalid_data(cleanup):
-    """Attempt to create a profile missing various data.
+
+@pytest.mark.parametrize('hosts, creds, profname',
+                         [  # bad credential
+                            (['localhost'], [-1], uuid4()),
+                          # bad host name
+                            (['*invalidhostname*'], None, uuid4()),
+                          # bad name
+                            (['localhost'], None, ''),
+                          ],
+                         ids=['bad-credential',
+                              'bad-hostname',
+                              'bad-name'])
+def test_negative_create_invalid_data(
+        cleanup,
+        shared_client,
+        hosts,
+        creds,
+        profname):
+    """Attempt to create a profile with invalid data.
 
     The requests should be met with a 4XX response.
 
     :id: e8754fd4-8d03-4899-bfde-0fc587d78ae1
     :description: Attempt to create profiles missing required data.
     :steps: Attempt to create a profile with invalid:
-            a) name
-            b) hosts
-            c) credential id's
+        a) creds
+        b) host
+        c) name
     :expectedresults: Error is thrown and no new profile is created.
-    :caseautomation: notautomated
+    :caseautomation: automated
     """
-    pass
+    cred = HostCredential(client=shared_client, password=uuid4())
+    cred.create()
+    cleanup.append(cred)
+    profile = NetworkProfile(
+        client=api.Client(response_handler=api.echo_handler),
+        hosts=hosts,
+        credential_ids=[cred._id] if not creds else creds,
+        name=profname,
+    )
+    create_response = profile.create()
+    assert create_response.status_code in range(400, 500)
+    read_response = profile.create()
```

Comment:
I think this is doing the same as above. Also you can't read what can't be created so checking for reading shouldn't be necessary.

Reply:
Yep, good call I got rid of these